### ISSUE TITLE
pytorch can't be as good as keras

### ISSUE THREAD

**Issue by mengjiexu** [pytorch/pytorch#18722]
I use the same data、 same data iter and same vgg16 network, but the pytorch's loss beame 12.3 while the keras's loss became 4.2:
pytorch loss log:
train Loss: 15.2963 val Loss: 8.0108
train Loss: 12.4626 val Loss: 9.5972
train Loss: 12.4143 val Loss: 6.9568
train Loss: 12.3993  val Loss: 8.3055
train Loss: 12.3956 val Loss: 6.8054

keras loss:
Epoch 1/20
402/402 [==============================] - 183s 455ms/step - loss: 11.6960 
Epoch 2/20
402/402 [==============================] - 181s 449ms/step - loss: 6.6525
Epoch 3/20
402/402 [==============================] - 181s 450ms/step - loss: 5.4451
Epoch 4/20
402/402 [==============================] - 186s 463ms/step - loss: 4.9515 
Epoch 5/20
402/402 [==============================] - 186s 462ms/step - loss: 4.6679 
Epoch 6/20
402/402 [==============================] - 186s 462ms/step - loss: 4.4298 

 The main code is as fallows:
pytorch:
model = nn.Sequential(*list(make_model('vgg16', num_classes=1, pretrained=True, input_size=INPUT_SIZE).children())[0][:],\
                      nn.AdaptiveAvgPool2d(1), 
                      Flatten(),nn.Sequential(nn.Linear(512, 1024, bias=True),nn.ReLU(), nn.Linear(1024, 1, bias=True))).cuda()
dataloaders = {'train':train_dl, 'val':valid_dl}
train_model(model, dataloaders, nn.L1Loss(), optim.Adam(model.parameters(),lr=0.001), 21)

keras:
base_model = VGG16(weights='imagenet', include_top=False, input_shape=(input_shape[1], input_shape[0], 3))
    model = GlobalAveragePooling2D()(base_model.output)
    model = Dense(1024, activation='relu')(model)
    model = Dense(1, activation='linear')(model)

    model = Model(inputs=base_model.input, outputs=model)
    # model = base_model

    keras.callbacks.EarlyStopping(
    monitor='val_loss', 
    patience=0, 
    verbose=0, 
    mode='auto')
    model.compile(keras.optimizers.Adam(0.001), 'mape', metrics=['acc', 'mae', 'mse'])
model.fit_generator(train_iter, len_train//BS-1, 20,validation_data=valid_iter, validation_steps=len_valid//BS, callbacks=[early_stop])

**Comment by xsacha** [comment]
Aren't the networks bound to be different and the pretrained models also bound to be different. Not sure what you can take from that.

**Comment by SsnL** [comment]
what's the code for `train_model`?

**Comment by ezyang** [comment]
There are any number of reasons why the training is diverging, and you haven't given us enough information to actually tell what your problem is. The best place to debug problems like this is in the forums: https://discuss.pytorch.org/
